app/models/manageiq/providers/redhat/infra_manager/inventory/strategies/v4.rb

```diff
@@ -107,12 +191,39 @@ def collect_vm_by_uuid(uuid)
       []
     end
 
+    def collect_templates_future
+      connection.system_service.templates_service.list(:wait => false)
+    end
+
+    def base_key_for_obj(obj, obj_type)
+      "#{obj_type}_#{obj.id}_"
+    end
+
+    def obj_attribute_request(obj, obj_type, attr_name)
+      objs_service = connection.system_service.send("#{obj_type}s_service")
+      procedure = proc do
+        objs_service.send("#{obj_type}_service", obj.id).send("#{attr_name}_service").list(:wait => false)
+      end
+      key = "#{base_key_for_obj(obj, obj_type)}#{attr_name}"
+      { key => procedure}
+    end
+
     def collect_templates
       connection.system_service.templates_service.list.collect do |template|
         TemplatePreloadedAttributesDecorator.new(template, connection, preloaded_disks)
       end
     end
 
+    def collect_future_template_disk_attachments(templates, templates_service)
+      templates_attachments = templates.collect do |template|
+        template_service = templates_service.template_service(template.id)
+        atts_service = template_service.disk_attachments_service
```

Comment:
`atts_service` isn't descriptive enough, can we name it `disk_attachments_service` and in correlation `disk_attachments_future`

Reply:
done, though I think its ok to have short names in small scopes